infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutKmPostDao.kt

```diff
@@ -189,6 +189,16 @@ class LayoutKmPostDao(
         }
     }
 
+    private fun getKmPostGkLocation(rs: ResultSet): TrackLayoutKmPostGkLocation? {
+        val location = rs.getGeometryPointOrNull("gk_point_x", "gk_point_y", "gk_srid")
+        val locationSource = rs.getEnumOrNull<KmPostGkLocationSource>("gk_location_source")
+        val locationConfirmed = rs.getBoolean("gk_location_confirmed")
+
+        return if (location != null && locationSource != null)
+            TrackLayoutKmPostGkLocation(location, locationSource, locationConfirmed)
+        else null
```

Comment:
Käytännössä gk-sijainti vaatii tästä eteenpäin sekä varsinaisen koordinaatin että tiedon siitä mitä kautta se tulee. Jos jompikumpi puuttuu, käsitellään kilometripylvästä niinkuin sillä ei olisi sijaintia.